lib/core/src/chain_swap.rs

```diff
@@ -377,23 +378,59 @@ impl ChainSwapHandler {
     }
 
     async fn handle_amountless_update(&self, swap: &ChainSwap) -> Result<(), PaymentError> {
+        let id = swap.id.clone();
+
+        // Since we optimistically persist the accepted receiver amount, if accepting a quote with
+        //  the swapper fails, we might still think it's accepted, so now we should get rid of the
+        //  old invalid accepted amount.
+        if swap.accepted_receiver_amount_sat.is_some() {
+            info!("Handling amountless update for swap {id} with existing accepted receiver amount. Erasing the accepted amount now...");
+            self.persister.update_accepted_receiver_amount(&id, None)?;
+        }
+
         let quote = self
             .swapper
-            .get_zero_amount_chain_swap_quote(&swap.id)
+            .get_zero_amount_chain_swap_quote(&id)
             .map(|quote| quote.to_sat())?;
-        info!("Got quote of {quote} sat for swap {}", &swap.id);
-
-        self.validate_and_update_amountless_swap(swap, quote)
-            .await?;
-        self.swapper
-            .accept_zero_amount_chain_swap_quote(&swap.id, quote)
+        info!("Got quote of {quote} sat for swap {}", &id);
+
+        match self.validate_amountless_swap(swap, quote).await? {
```

Comment:
I don't want the PR to change more, but for clarity we should consider renaming from "amountless"/"zero amount", as it also applies to chains swaps where the lockup amount differs from the amount set when the swap is created.

Reply:
I agree the names need to change. In some areas, I already took that into account, but because I didn't want to fully support that use case in this PR, I thought we'd change the names in a follow-up. Do you think it's worth changing the names now?

Reply:
No, let's close off this PR and keep it in mind (or an issue 🙂)

Reply:
https://github.com/breez/breez-sdk-liquid/issues/632#issuecomment-2569013875